src/components/common/dashboard/Dashboard.vue

```diff
@@ -0,0 +1,324 @@
+
+            let mockedWatchedOps = [
+                {
+                    operationId: "FinishedMatriculation1",
+                    initiator: "MockUser4",
+                    initiatedTimestamp: "2011-10-05T14:48:00.000Z",
+                    lastModifiedTimestamp: "2020-10-05T16:48:00.000Z",
+                    transactionInfo: {
+                        contractName: UC4Identifier.CONTRACT_MATRICULATION,
+                        transactionName: UC4Identifier.TRANSACTION_ADD_MATRICULATION,
+                        parameters: "SS2020 Computer Science v3",
+                    },
+                    state: OperationStatus.FINISHED,
+                    reason: "",
+                    existingApprovals: {
+                        users: [],
+                        groups: ["Admin"],
+                    } as ApprovalList,
+                    missingApprovals: {
+                        users: [],
+                        groups: [],
+                    } as ApprovalList,
+                } as Operation,
+            ];
+
+            onBeforeMount(async () => {
+                await refresh();
+                await getEnrollmentId();
+                await getRole();
+            });
+
+            const isAdmin = computed(() => {
+                return role.value == Role.ADMIN;
+            });
+
+            async function refresh() {
+                busy.value++;
+                store.commit(MutationTypes.CLEAR_TREATED_OPERATIONS);
+                await getOperations();
+                busy.value--;
+            }
+
+            async function getRole() {
+                busy.value++;
+                role.value = (await store.getters.user).role;
+                busy.value--;
+            }
+
+            async function getEnrollmentId() {
+                busy.value++;
+                const certificateManagement = new CertificateManagement();
+                const handler = new GenericResponseHandler("enrollment-id");
+                const response = await certificateManagement.getOwnEnrollmentId();
+                const result = handler.handleResponse(response);
+                if (result) {
+                    enrollmentId.value = result.id;
+                }
+                busy.value--;
+            }
+
+            function filterOperations(ops: Operation[], filter: string): Operation[] {
+                if (filter != "") {
+                    //TODO more filtering
+                    let filteredOperations = ops.filter(
+                        (op) =>
+                            op.operationId.replace(/\s/g, "").toLowerCase().includes(filter) ||
+                            op.initiator.replace(/\s/g, "").toLowerCase().includes(filter) ||
+                            op.transactionInfo.parameters.toString().replace(/\s/g, "").toLowerCase().includes(filter)
+                    );
+                    return filteredOperations;
+                }
+                return ops;
+            }
```

Comment:
imo this function could be extracted to a use/filterOperations, so we can easily add different filters there

Reply:
Good idea! Implemented with https://github.com/upb-uc4/ui-web/pull/785/commits/22c720333781a674e592c2b50afe7d9822cf19ea